src/common/configuration.ts

```diff
@@ -0,0 +1,72 @@
+import { ConfigurationChangeEvent, WorkspaceConfiguration } from 'vscode'
+import { ScanEngine } from './constants'
+
+export const CONFIGURATION_IDENTIFIER = 'spectral'
+export const USE_IAC_ENGINE_SETTING = 'scan.engines.useIacEngine'
+export const USE_OSS_ENGINE_SETTING = 'scan.engines.useOssEngine'
+export const USE_SECRET_ENGINE_SETTING = 'scan.engines.useSecretsEngine'
+export const INCLUDE_TAGS_SETTING = 'scan.includeTags'
+
+const prefixIdentifier = (settings: Array<string>) =>
+  settings.map((setting) => `${CONFIGURATION_IDENTIFIER}.${setting}`)
+
+export class Configuration {
+  private extensionConfig: WorkspaceConfiguration
+  static instance: Configuration
+  constructor(workspace) {
+    this.extensionConfig = workspace.getConfiguration(CONFIGURATION_IDENTIFIER)
+    workspace.onDidChangeConfiguration(
```

Comment:
this should be on `src/spectral/extension.ts` in `registerEvents` function

Reply:
fixed, https://github.com/SpectralOps/vscode-extension/pull/7/commits/e1fded33e247a3fd2e4a7e247f1b8c1868623af6